tests/main/snapd-reexec/task.yaml

```diff
@@ -0,0 +1,78 @@
+summary: Test that snapd rexecs itself into core
+
+systems: [-ubuntu-core-16-64, -ubuntu-core-16-arm-64, -ubuntu-core-16-arm-32]
+
+environment:
+    SNAPD_CONF: /etc/systemd/system/snapd.service.d/local.conf 
+
+execute: |
+    if [ "${SNAP_REEXEC:-}" = "0" ]; then
+        echo "skipping test when SNAP_REEXEC is disabled"
+        exit 0
+    fi
+
+    echo "Ensure we re-exec by default"
+    snap list
+    journalctl | MATCH "DEBUG: restarting into"
+
+    echo "Ensure that we do not re-exec into older versions"
+    systemctl stop snapd.service snapd.socket
+    echo "mount something older than our freshly build snapd"
+    mount --bind /bin/true /snap/core/current/usr/lib/snapd/snapd
+    systemctl start snapd.service snapd.socket
+    snap list
+    journalctl | MATCH "not restarting into.*older than"
+
+    echo "Revert back to normal"
+    systemctl stop snapd.service snapd.socket
+    umount /snap/core/current/usr/lib/snapd/snapd
```

Comment:
we probably somehow want to do a best effort at this umount also in  restore stanza

bit worried than when this test fail it will leave a strange state around